### PR TITLE
Fix adding parentheses after link removes link

### PR DESCRIPTION
**Pull request by eh2077**
### Fixed Issues
$ https://github.com/Expensify/App/issues/28344

# Tests
1. Go to a chat and add a link, like `https://staging.new.expensify.com/details`
2. Click the to edit the comment and add an extra parentheses `)` at the end of the link markdown, like
```
[Staging Detail](https://staging.new.expensify.com/details))
```
3. Save the draft
4. Verify link is not removed and the extra ending parentheses is not part of link and is displayed as plain text.

https://github.com/Expensify/expensify-common/assets/117511920/5caa4b3e-165c-4243-8d88-2a57d5dc884b

# QA
Same as tests.

### CODE REVIEW

File: lib/ExpensiMark.js [hunk 1]
```diff
@@ -766,11 +766,11 @@ export default class ExpensiMark {
      */
     extractLinksInMarkdownComment(comment) {
         try {
-            const escapedComment = _.escape(comment);
-            const matches = [...escapedComment.matchAll(MARKDOWN_LINK_REGEX)];
+            const html = this.replace(comment);
+            const matches = [...html.matchAll(new RegExp(`<a href="${MARKDOWN_URL_REGEX}" target="_blank" rel="noreferrer noopener">`, 'gi'))];
```
[the remark this answers is not in the document]
Comment (eh2077) [reply lib/ExpensiMark.js:770]: Yes, that's a good point. I think we should share the template of anchor tag if it's being changed often. But actually we don't change the format of `a` tag often, so I think it's acceptable to hard code it here. Having a sharing template for them seems a little bit over-engineering?

File: lib/ExpensiMark.js [hunk 2]
```diff
@@ -766,11 +766,11 @@ export default class ExpensiMark {
      */
     extractLinksInMarkdownComment(comment) {
         try {
-            const escapedComment = _.escape(comment);
-            const matches = [...escapedComment.matchAll(MARKDOWN_LINK_REGEX)];
+            const html = this.replace(comment);
```
[the remark this answers is not in the document]
Comment (eh2077) [reply lib/ExpensiMark.js]: Updated

File: lib/ExpensiMark.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (eh2077) [reply lib/ExpensiMark.js]: Added comment for the regex.

File: lib/ExpensiMark.js [hunk 3]
```diff
@@ -766,8 +766,11 @@ export default class ExpensiMark {
      */
     extractLinksInMarkdownComment(comment) {
         try {
-            const html = this.replace(comment);
-            const matches = [...html.matchAll(new RegExp(`<a href="${MARKDOWN_URL_REGEX}" target="_blank" rel="noreferrer noopener">`, 'gi'))];
+            const htmlString = this.replace(comment);
+
+            // We use same anchore tag template as link and autolink rules to extract link
```
Comment (lakchote) [inline lib/ExpensiMark.js:771]: @eh2077 Can you please fix the typo so that `anchore` becomes `anchor`? Thanks

**Review by lakchote — CHANGES_REQUESTED** [review]

File: lib/ExpensiMark.js (hunk 3 above)
Comment (eh2077) [reply lib/ExpensiMark.js]: @lakchote Updated

Comment (lakchote) [reply lib/ExpensiMark.js]: Thanks!

**Review by lakchote — APPROVED** [review]

**Pull request merged**